test-data/unit/check-partially-defined.test

```diff
@@ -319,3 +319,20 @@ def f() -> None:
         fail()
     z = y  # E: Name "y" may be undefined
     z = x
+
+[case testDictComprehension]
+# flags: --enable-error-code partially-defined
+
+def f() -> None:
+    for _ in some_iterator():  # type: ignore[name-defined]
```

Comment:
nit: maybe just do `for _ in [1, 2, 3]` and you get to avoid the extra type ignore/